### PR TITLE
Make size parameter on 'replicate' explicit

### LINKED ISSUE #2097: `n` in `Vec.replicate` should be explicit
> > > > I agree. The argument to Vec.iterate should be explicit, not implicit
> > >
> > > Should that be the case with `Vec.replicate` too?
> >
> >Yes, nice spot!
>
> Turns out, Vec.replicate is comparatively a lot more involved within the codebase. Would it be better if I create a downstream PR for this (given that I have updated 15+ files till now, and there still are files left with Vec.replicate occurrences)?

_Originally posted in https://github.com/agda/agda-stdlib/issues/2051#issuecomment-1685420148_

(creating an issue to ensure that this isn't lost between PRs and reviews)

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
Fixes #2097. Lots of implicits being passed explicitly removed, so it looks like it was the right idea :tada:

### CODE REVIEW

**Comment by jamesmckinna** [comment]
I'll try to review tomorrow.

File: src/Data/Vec/Functional/Relation/Unary/All/Properties.agda [hunk 1]
```diff
@@ -21,79 +22,72 @@ open import Relation.Unary
 private
   variable
     a b c p q r ℓ : Level
-    A : Set a
-    B : Set b
-    C : Set c
+    A B C : Set a
```
Comment (jamesmckinna) [inline src/Data/Vec/Functional/Relation/Unary/All/Properties.agda:25]: In view of your refactoring of the declarations for `A B C`, do we need so many `Levels` declared as well? Or would `a p ℓ : Level` suffice?

**Comment by jamesmckinna** [comment]
The extensive `variable` refactoring is nice, but see my comment.
Agree that the issue was a good one to pursue, given the number of occurrences!

File: src/Data/Vec/Functional/Relation/Unary/All/Properties.agda (hunk 1 above)
Comment (MatthewDaggitt) [reply src/Data/Vec/Functional/Relation/Unary/All/Properties.agda]: Yup, nice spot!

**Pull request merged**